### ISSUE TITLE
how to finetune with a pretrained model

### ISSUE THREAD

**Issue by Johnson-yue** [open-mmlab/mmdetection#2669]
Hi， I see the Tutorial1 finetune but I am still confuse!!

Q1: I can inhert base configs, but I don't know what I should modify??
I know I can modify head w.r.t my new dataset, it make sense.

the dataset ?? Maybe I  should modify data_root ?? or ann_files ??

the optimizer,  which type is supported except 'SGD' ??
the lr_config , if I do not want to use warmup , what should I do ??
those just for multi-GPU， but I only have one

the default_runtime.py
what is meaning for 【workflow】
Could I modify the 'train' to 'finetune'??? which type is supported ?

Q2: The Tutorial is piece by piece， so I don‘t know if I finetune  the model what Everything I need to do。 I only know a piece and other piece

I trying to finetune a retinanet but maybe it is new。。。。So I am still working

**Comment by ZwwWayne** [comment]
Hi @Johnson-yue , 
1. You need to change to use your own dataset, and you can find how to use your own dataset on other [tutorial](https://mmdetection.readthedocs.io/en/latest/tutorials/new_dataset.html).
2. All the optimizers supported by PyTorch is supported, the detail can also be found in this [tutorial](https://mmdetection.readthedocs.io/en/latest/tutorials/new_modules.html#customize-optimizer).

The document in this PR https://github.com/open-mmlab/mmdetection/pull/2645 might do some help to you. It will present a full config of mask rcnn with comments. We will try to merge it ASAP.

**Comment by Johnson-yue** [comment]
@ZwwWayne 

   1. My question is how to modify the config, make dataset is no problem( I used coco format , for simple cost)
   2.I want to know what are supported now, so that I can choose one from the existing ones instead of writing a new one myself.  This [tutorial](https://mmdetection.readthedocs.io/en/latest/tutorials/new_modules.html#customize-optimizer) is not an answer, but I will try it for later.
   3. [this PR](https://github.com/open-mmlab/mmdetection/pull/2645) likely I want to see. But I still confused about some parameters meaning , such as **workflow**, what is meaning ?  

It seems that you told me just use it , but not told me what is it , why you are using this 

Is there document have an introduce for all keyword that in config

**Comment by ZwwWayne** [comment]
> My question is how to modify the config

This [tutorial](https://mmdetection.readthedocs.io/en/latest/tutorials/new_dataset.html) should already tell you how to modify the config of the dataset.

> want to know what is supported now, so that I can choose one from the existing ones instead of writing a new one myself.

Answered: `All the optimizers supported by PyTorch is supported`. Is there any more you want to use or do not know how/where to find it?

>But I still confused about some parameters meaning, such as workflow...

The workflow allows key `train` and `val`, and (train, 1) means run training process for one time, more details can be found [here](https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/runner.py). 
It is designed for more complicated workflow, though mmdet's workflow is pretty simple.
Usually, we do not need to change the workflow. If you have special needs and do not know how to modify it, feel free to ask.

> It seems that you told me just use it, but not told me what is it, why you are using this.
The hyperparameters 

Most of the hyperparameter is set according to the original paper or the authors' implementation, so if you have questions, you should read the related papers. Are there any other parameters you are confused about? 

>Is there document has an introduction for all keyword that in config

This PR https://github.com/open-mmlab/mmdetection/pull/2645 aims to do that, feel free if you have any more need.

**Comment by Johnson-yue** [comment]
Thank you， I can finetune it as your said， I have no question 。I will hack it more and do more experiement
